Modules/@babylonjs/react-native/windows/BabylonReactNative/EngineView.cpp

```diff
@@ -2,17 +2,38 @@
 #include "EngineView.h"
 #include "EngineView.g.cpp"
 
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::System::Threading;
+using namespace winrt::Windows::UI::Core;
 using namespace winrt::Windows::UI::Xaml;
 using namespace winrt::Windows::UI::Xaml::Input;
 using namespace winrt::Windows::UI::Xaml::Media;
 using namespace winrt::Windows::UI::Xaml::Controls;
 
 namespace winrt::BabylonReactNative::implementation {
-    EngineView::EngineView() {
+	EngineView::EngineView() {
+
         _revokerData.SizeChangedRevoker = SizeChanged(winrt::auto_revoke, { this, &EngineView::OnSizeChanged });
-        _revokerData.PointerPressedRevoker = PointerPressed(winrt::auto_revoke, { this, &EngineView::OnPointerPressed });
-        _revokerData.PointerMovedRevoker = PointerMoved(winrt::auto_revoke, { this, &EngineView::OnPointerMoved });
-        _revokerData.PointerReleasedRevoker = PointerReleased(winrt::auto_revoke, { this, &EngineView::OnPointerReleased });
+
+        WorkItemHandler workItemHandler([weakThis{ this->get_weak() }](IAsyncAction const& /* action */)
+        {
+            if (auto trueThis = weakThis.get())
+            {
+                auto deviceTypes = static_cast<CoreInputDeviceTypes>(
+                    static_cast<uint32_t>(Windows::UI::Core::CoreInputDeviceTypes::Mouse) |
+                    static_cast<uint32_t>(Windows::UI::Core::CoreInputDeviceTypes::Touch) |
+                    static_cast<uint32_t>(Windows::UI::Core::CoreInputDeviceTypes::Pen));

```

Comment:
The Babylon.js `DeviceSourceManager` distinguishes between these (at least touch and mouse - see https://github.com/BabylonJS/Babylon.js/blob/master/src/DeviceInput/InputDevices/deviceTypes.ts). `NativeInput` is currently hard coded to always raise events for `Touch` only (that's just as much as has been implemented). I think what would probably make more sense would be to restrict this code to handle `Touch` only, or do more plumbing in `NativeInput` to support additional device types. If you want to try doing the latter, let me know and I'm happy to discuss.

Reply:
I am pretty sure if we only support touch none of this logic will work for x64 desktop scenarios with a mouse. whats the suggestion for pen? having it register as touch?

Reply:
FWIW, input does not work currently. This change fixes that bug. It sounds like updating NativeInput involves feature work compared to bug fixes.

Reply:
Yes, no work had previously been done to support mouse in `NativeInput`, so there is "feature" level work needed to support it correctly.